api/pom.xml

```diff
@@ -133,6 +133,14 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-release-plugin</artifactId>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>1.7</source>
```

Comment:
Discussed offline with @drcrallen and this is apparently due to an abundance of caution about working with existing extensions. Although, I think this abundance of caution is not abundant enough, since we have extension points like AggregatorFactory that are not in druid-api (see http://druid.io/docs/latest/development/modules.html).

@drcrallen @leventov do you know if it would be enough to do the following:

1. Compile everything (including druid-api) for java 1.8 starting in 0.10.1
2. Add to the release notes for 0.10.0 "extensions must target java 1.8 to be compatible with future 0.10.x versions"
3. Expect that users that follow the above guideline will be able to build an extension for 0.10.0 and have it work with 0.10.1.

If so then this change LGTM